src/main/scala/dpla/ingestion3/model/RowConverter.scala

```diff
@@ -16,103 +17,102 @@ import org.apache.spark.sql.types.{ArrayType, StringType, StructField, StructTyp
+        fromEdmWebResource(oreAggregation.isShownAt), //6
+        oreAggregation.`object`.map(fromEdmWebResource).orNull, //7
+        oreAggregation.preview.map(fromEdmWebResource).orNull, //8
+        fromEdmAgent(oreAggregation.provider), //9
+        oreAggregation.edmRights.map(_.toString).orNull //10
+      ),
       sqlSchema
     )
+  }
 
-  private[model] def oreAggregation(oa: OreAggregation): Row = Row(
-    oa.uri.toString,
-    edmAgent(oa.dataProvider),
-    oa.originalRecord,
-    oa.hasView.map(edmWebResource),
-    oa.intermediateProvider.map(edmAgent).orNull,
-    oa.`object`.map(edmWebResource).orNull,
-    oa.preview.map(edmWebResource).orNull,
-    edmAgent(oa.provider),
-    oa.edmRights.map(_.toString).orNull
-  )
-
-  private[model] def dplaSourceResource(sr: DplaSourceResource): Row = Row(
-    sr.alternateTitle,
-    sr.collection.map(dcmiTypeCollection),
-    sr.contributor.map(edmAgent),
-    sr.creator.map(edmAgent),
-    sr.date.map(edmTimeSpan),
-    sr.description,
-    sr.extent,
-    sr.format,
-    sr.genre.map(skosConcept),
-    sr.identifier,
-    sr.language.map(skosConcept),
-    sr.place.map(dplaPlace),
-    sr.publisher.map(edmAgent),
-    sr.relation.map(literalOrUri),
-    sr.replacedBy,
-    sr.replaces,
-    sr.rights,
-    sr.rightsHolder.map(edmAgent),
-    sr.subject.map(skosConcept),
-    sr.temporal.map(edmTimeSpan),
-    sr.title,
-    sr.`type`
+  private[model] def fromEdmWebResource(wr: EdmWebResource): Row = Row(
+    wr.uri.toString, //0
+    wr.fileFormat, //1
+    wr.dcRights, //2
+    wr.edmRights.orNull //3
   )
 
-  private[model]def edmWebResource(wr: EdmWebResource): Row = Row(
-    wr.uri.toString,
-    wr.fileFormat,
-    wr.dcRights,
-    wr.edmRights.orNull
+  private[model] def fromSourceResource(sr: DplaSourceResource): Row = Row(
+    sr.alternateTitle, //0
+    sr.collection.map(fromDcmiTypeCollection), //1
+    sr.contributor.map(fromEdmAgent), //2
+    sr.creator.map(fromEdmAgent), //3
+    sr.date.map(fromEdmTimeSpan), //4
+    sr.description, //5
+    sr.extent, //6
+    sr.format, //7
+    sr.genre.map(fromSkosConcept), //8
+    sr.identifier, //9
+    sr.language.map(fromSkosConcept), //10
+    sr.place.map(fromDplaPlace), //11
+    sr.publisher.map(fromEdmAgent), //12
+    sr.relation.map(fromLiteralOrUri), //13
+    sr.replacedBy, //14
+    sr.replaces, //15
+    sr.rights, //16
+    sr.rightsHolder.map(fromEdmAgent), //17
+    sr.subject.map(fromSkosConcept), //18
+    sr.temporal.map(fromEdmTimeSpan), //19
+    sr.title, //20
+    sr.`type` //21
   )
 
-  private[model] def edmAgent(ea: EdmAgent): Row = Row(
-    ea.uri.map(_.toString).orNull,
-    ea.name.orNull,
-    ea.providedLabel.orNull,
-    ea.note.orNull,
-    ea.scheme.map(_.toString).orNull,
-    ea.exactMatch.map(_.toString),
-    ea.closeMatch.map(_.toString)
+  private[model] def fromEdmAgent(ea: EdmAgent): Row = Row(
+    ea.uri.map(_.toString).orNull, //0
```

Comment:
Thank you for commenting the array indices here. Future Scott owes you a beer for this stroke of genius.